src/pocketmine/utils/TextFormat.php

```diff
@@ -80,6 +80,18 @@ public static function clean(string $string, bool $removeFormat = true) : string
 		return str_replace("\x1b", "", preg_replace("/\x1b[\\(\\][[0-9;\\[\\(]+[Bm]/", "", $string));
 	}
 
+	/**
+	 * Replaces placeholders of § with the correct character. Only valid codes (as in the constants of the TextFormat class) will be converted.
+	 *
+	 * @param string $string
+	 * @param string $placeholder default "&"
+	 *
+	 * @return string
+	 */
+	public static function colorize(string $string, string $placeholder = "&") : string{
+		return preg_replace('/' . preg_quote($placeholder) . '([0-9a-fk-or])/u', TextFormat::ESCAPE . '$1', $string);
```

Comment:
at least this regex is better than https://github.com/pmmp/PocketMine-MP/pull/1837/files#diff-b1b8c79c93533368678816cecafcbe98R65